### PR TITLE
(refactor) Service Queues refactoring to consolidate useQueues hook

### PR DESCRIPTION
**Pull request by chibongho**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Summary

We have a hook redefined in multiple places to fetch a list of queues from the backend. This PR consolidates them. Also renamed it to `useQueues` instead of `useServices` as a more accurate description of what it is.

Tested manually by making sure list of queues are still loading fine in various dropdowns in:
- "Add patient to queue" form
- ClinicMetrics component
- "Show patients waiting for" dropdown

### CODE REVIEW

**Review by mseaton — COMMENTED** [review]
LGTM. Will defer to others to approve.

File: packages/esm-service-queues-app/src/helpers/useQueues.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import useSWRImmutable from 'swr/immutable';
+import { type QueueServiceInfo } from '../types';
+
+export function useQueues(locationUuid?: string) {
+  const apiUrl = '/ws/rest/v1/queue' + (locationUuid ? `?location=${locationUuid}` : '');
+
+  const { data } = useSWRImmutable<{ data: { results: Array<QueueServiceInfo> } }, Error>(apiUrl, openmrsFetch);
```
Comment (denniskigen) [inline packages/esm-service-queues-app/src/helpers/useQueues.ts:8]: Do we expect queues to be [immutable](https://swr.vercel.app/docs/revalidation.en-US#disable-automatic-revalidations) resources? If not, you could consider enabling automatic revalidations for the resource by using the useSWR hook instead.

**Review by denniskigen — APPROVED** [review]
Thanks, @chibongho. LGTM. As Mark said, I'd defer to others to merge. Just one concern about whether the queue resource is immutable or not. Also, probably best to add a conventional commit label to your PR title. I'd recommend using `(refactor)`.

File: packages/esm-service-queues-app/src/types/index.ts [hunk 2]
```diff
@@ -291,11 +291,23 @@ export interface QueueEntryPayload {
   };
 }
 
+export interface AllowedPriority {
```
Comment (CynthiaKamau) [inline packages/esm-service-queues-app/src/types/index.ts:294]: Why not use `OpenmrsResource ` ?

**Review by CynthiaKamau — COMMENTED** [review]
LGTM 🚀 , thanks @chibongho

File: packages/esm-service-queues-app/src/types/index.ts (hunk 2 above)
Comment (chibongho) [reply packages/esm-service-queues-app/src/types/index.ts:294]: Ahh... didn't know that exists. I think explicitly naming the type is still a good idea, but I'll do `type AllowedPriority = OpenmrsResource` instead.

File: packages/esm-service-queues-app/src/helpers/useQueues.ts (hunk 1 above)
Comment (chibongho) [reply packages/esm-service-queues-app/src/helpers/useQueues.ts:8]: I believe so. Queues almost never change.

**Pull request merged**